contracts/erc721/InvoiceUnpaidNFT.sol

```diff
@@ -196,64 +176,85 @@ contract InvoiceUnpaidNFT is Initializable, UserMintableERC721 {
       signingRoot_,
       INVOICE_SENDER,
       _getSender(),
-      salts[SENDER_IDX],
-      proofs[SENDER_IDX]
+      pd.salts[SENDER_IDX],
+      pd.proofs[SENDER_IDX]
+    );
+
+    // Extract the public key from the signature
+    bytes32 pbKey_ = bytes32(
+      uint256(
+        Signatures.consensusSignatureToEthSignedMessageHash(signingRoot_, pd.values[SIGNATURE_TRANSITION_IDX][0]).recover(pd.values[SIGNATURE_IDX]))
```

Comment:
We have our own implementation of the simple encoding to not have to hash the contents.